src/pages/ImportData.js

```diff
@@ -30,36 +33,84 @@ const styles = {
   }
 }
 
-const ChoiceButton = ({ iconName, title, detail, style, ...props }) => {
+const ChoiceButton = ({ onClick, iconName, title, detail, tooltip }) => {
   return h(Clickable, {
+    onClick,
+    tooltip,
     style: {
-      ...style,
-      padding: '1rem',
+      padding: '1rem', marginTop: '1rem',
       display: 'flex', alignItems: 'center',
-      border: `1px solid ${colors.accent()}`, borderRadius: 4
+      border: `1px solid ${colors.accent(1)}`, borderRadius: 4
     },
-    hover: { backgroundColor: colors.accent(0.1) },
-    ...props
+    hover: { backgroundColor: colors.accent(0.1) }
   }, [
-    icon(iconName, { size: 24, style: { flex: 'none', marginRight: '1rem' } }),
+    icon(iconName, { size: 29, style: { flex: 'none', marginRight: '1rem', color: colors.accent(1) } }),
     div({ style: { flex: 1 } }, [
-      div({ style: { fontWeight: 'bold' } }, [title]),
+      div({ style: { fontWeight: 'bold', color: colors.accent(1) } }, [title]),
       div([detail])
     ]),
-    icon('angle-right', { size: 32, style: { flex: 'none', marginLeft: '1rem' } })
+    icon('angle-right', { size: 32, style: { flex: 'none', marginLeft: '1rem', color: colors.accent(1) } })
   ])
 }
 
 const ImportData = () => {
   const { workspaces, refresh: refreshWorkspaces } = useWorkspaces()
   const [isImporting, setIsImporting] = useState(false)
-  const { query: { url, format, ad, wid } } = Nav.useRoute()
+  const { query: { url, format, ad, wid, template } } = Nav.useRoute()
   const [mode, setMode] = useState(wid ? 'existing' : undefined)
   const [isCreateOpen, setIsCreateOpen] = useState(false)
+  const [isCloneOpen, setIsCloneOpen] = useState(false)
   const [selectedWorkspaceId, setSelectedWorkspaceId] = useState(wid)
+  const [selectedTemplateWorkspace, setSelectedTemplateWorkspace] = useState()
+  const [radioButtonSelected, setRadioButtonSelected] = useState()
+  const [templateWorkspaces, setTemplateWorkspaces] = useState()
+
+  const noteMessage = 'Note that the import process may take some time after you are redirected into your destination workspace.'
 
   const selectedWorkspace = _.find({ workspace: { workspaceId: selectedWorkspaceId } }, workspaces)
 
+  const getTemplates = fileContents => {
+    let temporaryList = []
+
+    !(_.isArray(template)) ? temporaryList = fileContents[template] :
+      _.forEach(currentTemplate => {
+        fileContents[currentTemplate] &&
+        temporaryList.push(fileContents[currentTemplate])
```

Comment:
There's a way to do this computation using lodash helpers and eliminate the mutation. Let's pair on it.